File: output/qwen_2.5_72b_instruct/codenet/direct_translation/python/java/successful_translation/s200092246.java
import java.util.Scanner;

public class s200092246 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int[] ans = new int[2];

        for (int i = 1; i <= N; i++) {
            int sum = (i * (i + 1)) / 2;
            if (sum < N) {
                continue;
            } else {
                ans[0] = i - 1;
                ans[1] = sum - N;
                break;
            }
        }
        for (int i = 0; i <= ans[0]; i++) {
            if (i + 1 != ans[1]) {
                System.out.println(i + 1);
            }
        }
    }
}

File: dataset/codenet/Python/Code/s200092246.py
N = int(input())
ans = []

for i in range(1,N+1):
    if sum(range(1, i + 1)) < N:
        continue
    else:
        ans = [i - 1,sum(range(1, i + 1)) - N]
        break
for i in range(ans[0]+1):
    if i + 1 != ans[1]:
        print(i + 1)
